src/v/wasm/tests/wasm_transform_test.cc

```diff
@@ -106,3 +108,16 @@ TEST_F(WasmTestFixture, MemoryIsLimited) {
     EXPECT_NO_THROW(execute_command("allocate", 5_KiB));
     EXPECT_THROW(execute_command("allocate", MAX_MEMORY), wasm::wasm_exception);
 }
+
+TEST_F(WasmTestFixture, CpuIsLimited) {
+    load_wasm("dynamic.wasm");
+    // In reality, we don't want this to be over a few milliseconds, but in an
+    // effort to prevent test flakiness on hosts with lots of concurrent tests
+    // on shared vCPUs, we make this much higher.
+    ss::engine().update_blocked_reactor_notify_ms(50ms);
+    bool stalled = false;
+    ss::engine().set_stall_detector_report_function(
+      [&stalled] { stalled = true; });
+    EXPECT_THROW(execute_command("loop", 0), wasm::wasm_exception);
+    EXPECT_FALSE(stalled);
```

Comment:
does the opposite case work too? experience a stall with 1ms notify threshold?

Reply:
Yes it does, although I suppose that is potentially flaky as well. Do you think we should have a test for that?

Reply:
no, not if you already tried it